packages/main/src/Slider.js

```diff
@@ -0,0 +1,292 @@
+ * @public
+ */
+class Slider extends UI5Element {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return SliderCss;
+	}
+
+	static get template() {
+		return SliderTemplate;
+	}
+
+	static async define(...params) {
+		super.define(...params);
+	}
+
+	constructor() {
+		super();
+
+		this._items = ["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16"];
+		this.value = this._items[0];
+		//this.i18nBundle = getI18nBundle("@ui5/webcomponents");
+		this._scroller = new ScrollEnablement(this);
+		this.prevElement = 0;
+		this._currentElementIndex = 0;
+		this._itemCellHeight = 0;
+	}
+
+	_findSelectedElement(){
+		let itemsList = this.shadowRoot.querySelector(`#${this._id}--items-list`),
+			parentOffset = itemsList.parentElement.parentElement.offsetTop,
+			itemsListArray = [].slice.call(itemsList.children),
+			firstVisibleElementIndex = 0;
+
+		while ((itemsListArray[firstVisibleElementIndex].getBoundingClientRect().y - parentOffset) < 0){
+			firstVisibleElementIndex++;
+		}
+
+		return itemsListArray[firstVisibleElementIndex + 5];
+	}
+
+	_updateScrolling(e){
+		let sizeInRems = this._items.length * 3, // the size of one element in rems (16px = 1rem)
+			sizeOfOneElementInPixels = _itemCellHeight * 16,
+			indexForOffset;
+		const elements = this.shadowRoot.querySelectorAll(".sapMWSItem");
+
+		if (!this._findSelectedElement()){
+			return;
+		}
+		if (this.value === this._findSelectedElement().textContent) {
+			return;
+		}
+
+		if ( (e.scroll / sizeOfOneElementInPixels) / 0.5 > 1)  {
+			indexForOffset = Math.ceil(e.scroll / sizeOfOneElementInPixels);
+		} else {
+			indexForOffset = Math.floor(e.scroll / sizeOfOneElementInPixels);
+		}
+
+		this._selectElement(this._findSelectedElement());
+		this.value = this._findSelectedElement().textContent;
+	}
+
+	onAfterRendering() {
+		this._scroller.scrollContainer = this.shadowRoot.querySelector(`#${this._id}--wrapper`);
+		this._scroller.attachEvent("scroll", this._updateScrolling.bind(this));
+		if (this.expanded) {
+			const elements = this.shadowRoot.querySelectorAll(".sapMWSItem");
+			this._itemCellHeight = elements[0].offsetHeight / 16;
+			for (let i = 0; i < elements.length; i++){
+				if (elements[i].textContent === this.value){
+					this._selectElement(elements[i]);
+					return true;
+				}
+			}
+
+			this._selectElement(elements[0]);
+		}
+	}
+
+	get items() {;
+		return this._items;
+	}
+
+	get isExpandedClass(){
+		if (this.expanded){
+			return " sapMWSExpanded";
+		}
+
+		return "";
+	}
+
+	get isDisabledClass(){
+		if (this.disabled){
+			return " sapMWSDisabled";
+		}
+
+		return "";
+	}
+
+	_onelementclick(e) {
+		if(e.target.classList.contains("sapMWSItem") && this.expanded) {
+			this.value = e.target.textContent;
+			this._selectElement(e.target);
+		}
+
+		if (!this.expanded && e.target.classList.contains("sapMWSItem")) {
+			this.expanded = true;
+			this._invalidate();
```

Comment:
You can't call this method. If you need to invalidate, change the state. If you can't invalidate by changing the state, then you have "hidden state". In that case, declare the variable that needs to invalidate in the metadata as a private property.